Kiosk/Views/InkRecognizerExplorer/DualCanvas.xaml.cs

```diff
@@ -0,0 +1,860 @@
+                }
+
+                clearedStrokes.Clear();
+                inkCleared = false;
+            }
+            else if (activeTool is InkToolbarEraserButton)
+            {
+                RedoButton_Click(null, null);
+            }
+            else
+            {
+                var strokes = inkCanvas.InkPresenter.StrokeContainer.GetStrokes();
+                if (strokes.Count > 0)
+                {
+                    var stroke = strokes[strokes.Count - 1];
+
+                    redoStrokes.Push(stroke);
+
+                    stroke.Selected = true;
+                    inkCanvas.InkPresenter.StrokeContainer.DeleteSelected();
+                }
+            }
+        }
+
+        private void RedoButton_Click(object sender, RoutedEventArgs e)
+        {
+            var button = sender as InkToolbarCustomToolButton;
+            button.IsChecked = false;
+
+            if (redoStrokes.Count > 0)
+            {
+                var stroke = redoStrokes.Pop();
+
+                inkCanvas.InkPresenter.StrokeContainer.AddStroke(stroke.Clone());
+            }
+        }
+
+        private void ClearButton_Click(object sender, RoutedEventArgs e)
+        {
+            var button = sender as InkToolbarCustomToolButton;
+            button.IsChecked = false;
+
+            var strokes = inkCanvas.InkPresenter.StrokeContainer.GetStrokes();
+            foreach (var stroke in strokes)
+            {
+                clearedStrokes.Add(stroke);
+            }
+
+            inkCleared = true;
+            inkCanvas.InkPresenter.StrokeContainer.Clear();
+            ClearJson();
+        }
+
+        private void TouchButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (touchButton.IsChecked == true)
+            {
+                inkCanvas.InkPresenter.InputDeviceTypes |= CoreInputDeviceTypes.Touch;
+            }
+            else
+            {
+                inkCanvas.InkPresenter.InputDeviceTypes &= ~CoreInputDeviceTypes.Touch;
+            }
+        }
+        #endregion
+
+        #region Event Handlers - Ink Canvas
+        private void InkPresenter_StrokeInputStarted(InkStrokeInput sender, PointerEventArgs args)
+        {
+            ViewCanvasButton_Click(null, null);
+
+            clearedStrokes.Clear();
+            inkCleared = false;
+
+            activeTool = inkToolbar.ActiveTool;
+        }
+
+        private void InkPresenter_StrokesErased(InkPresenter sender, InkStrokesErasedEventArgs args)
+        {
+            // When strokes are erased they are treated the same as an undo and are pushed onto a stack of strokes for the redo button
+            foreach (var stroke in args.Strokes)
+            {
+                redoStrokes.Push(stroke);
+            }
+
+            activeTool = inkToolbar.ActiveTool;
+        }
+
+        private void LanguageDropdown_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            if (inkRecognizer != null)
```

Comment:
> if (inkRecognizer != null) [](start = 11, length = 27)

seems kind of random to check for null.  I'd remove it  or add checks everywhere if there is a chance this could be null. given that you are building it in the constructor I'm thinking you don't need to check..